src/theory/arith/arith_evaluator.cpp

```diff
@@ -0,0 +1,94 @@
+#include "theory/arith/arith_evaluator.h"
+
+#include "theory/arith/nl/poly_conversion.h"
+#include "theory/rewriter.h"
+#include "theory/theory.h"
+#include "util/real_algebraic_number.h"
+
+namespace cvc5 {
+namespace theory {
+namespace arith {
+
+namespace {
+
+RealAlgebraicNumber evaluate(TNode expr,
```

Comment:
Shouldn't we do a dag traversal for this? 
Also technically we should make this iterative not recursive.

Reply:
Theoretically, yes. However, we will soon replace this code by properly integrating RANs into the arithmetic rewriter...

Reply:
Ok thats fine for now then.